src/jdk.incubator.foreign/share/classes/jdk/incubator/foreign/LibraryLookup.java

```diff
@@ -48,14 +48,33 @@
  */
 public interface LibraryLookup {
 
+    /**
+     * A symbol retrieved during a library lookup. A lookup symbol has a <em>name</em> and can be projected
+     * into a memory address (see {@link #name()} and {@link #address()}, respectively).
+     */
+    interface Symbol extends Addressable {
```

Comment:
Sealed and implementations are values?
It might even make sense as a value-based record, perhaps a little early to tell right now.